File: r/perfect-numbers/perfect-numbers.R
number_type <- function(n){
<<<<<<< Updated upstream
  stopifnot(n > 0)
  numrange <- 1:(n %/% 2)
  aliquot <- sum(numrange[n %% numrange == 0])
  
  if (n == 1 || aliquot < n) return("deficient")
  if (aliquot == n) return("perfect")
  if (aliquot > n) return("abundant")
=======
  
>>>>>>> Stashed changes
}

File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """

    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")
    aliquot = sum(f for f in range(1, number // 2 + 1) if number % f == 0)
    if aliquot == number:
        return "perfect"
    if  aliquot < number:
        return "deficient"
    if  aliquot > number:
        return "abundant"
